### PR TITLE
(1) Enable hash server usage (2) Introduce key scheduler

### PR DESCRIPTION
**Pull request by chowderchow**
## Description
The KeyScheduler class in scheduler.py serves as the basic building
block for more complex scheduling in the future (throttling by ms, query
remaining hashes, etc). Similar to how BaseScheduler is used for more complex scan methods. 

Usage is similar to any args that are usually parsed as a list:
ex: `-hk KEY_1 -hk KEY_2 -hk KEY_3`

## Motivation and Context
Enables 0.51 API usage. 

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by hokiepokedad** [comment]
I'm assuming I can't submit a PR of a PR can I?   :)

**Comment by sclo012** [comment]
This needs to be merged as soon as possible so the people that wants to use .51 can start using it with multiple 150 keys.

**Comment by Skorfius** [comment]
@sclo012 u could always merge it on ur own if you want it that badly!

**Comment by hokiepokedad** [comment]
@chowderchow are you going to merge my Hash Key status page for -ps into this?

**Comment by chowderchow** [comment]
@hokiepokedad that will come after basic functionality is merged, though I'd prefer if you'd be the one to open a separate PR for that seeing as most of the code was yours.

**Comment by hokiepokedad** [comment]
@chowderchow sounds good

**Comment by tokar86a** [comment]
Hope we are going to see this one merge soon :)

**Comment by fufufabulous** [comment]
i get this when i want to use -hk

2017-01-08 11:39:04,543 [ search-worker-2][        search][   ERROR] Exception in search_worker under account kohblofstrap Exception message: .
Traceback (most recent call last):
  File "/root/PokemonGo-Map/pogom/search.py", line 602, in search_worker_thread
    api.activate_hash_server(key)
  File "/opt/python/lib/python2.7/site-packages/pgoapi/pgoapi.py", line 155, in __getattr__
    raise AttributeError
AttributeError

**Comment by zhomz** [comment]
I get the same exception as @fufufabulous

File: pogom/search.py [hunk 1]
```diff
@@ -592,6 +597,11 @@ def search_worker_thread(args, account_queue, account_failures, search_items_que
                 # when the auth token is refreshed.
                 api.set_position(*step_location)
 
+                if args.hash_key:
+                    key = key_scheduler.next()
+                    api.activate_hash_server(key)
+                    log.info('Using key {} for this scan.'.format(key))
```
Comment (smc87) [inline pogom/search.py]: Do we need to see this at info level? It strikes me as a debug sort of thing.

**Comment by fufufabulous** [comment]
Thanks it works now, how can we know the amount of requests/min ?

File: pogom/search.py [hunk 2]
```diff
@@ -304,6 +305,10 @@ def search_overseer_thread(args, new_location_queue, pause_bit, heartb, db_updat
         t.daemon = True
         t.start()
 
+    # Create the key scheduler.
+    if args.hash_key:
+        key_scheduler = schedulers.KeyScheduler(args.hash_key).scheduler()
+
```
Comment (smc87) [inline pogom/search.py]: If turning the other logs to debug, then maybe a single info message here to say that hash server is enabled?

**Pull request merged**